src/test/cuecontrol_test.cpp

```diff
@@ -266,35 +266,53 @@ TEST_F(CueControlTest, SeekOnLoadDefault) {
 }
 
 TEST_F(CueControlTest, SeekOnLoadMainCue) {
+    config()->set(ConfigKey("[Controls]", "CueRecall"),
+            ConfigValue(static_cast<int>(SeekOnLoadMode::MainCue)));
+    TrackPointer pTrack = createTestTrack();
+    loadTrack(pTrack);
+
+    // We expect a cue point at the very beginning
+    EXPECT_DOUBLE_EQ(0.0, m_pCuePoint->get());
+    EXPECT_DOUBLE_EQ(0.0, getCurrentSample());
+
+    // Move cue like silence analysis does and check if track is following it
+    pTrack->setCuePoint(CuePosition(200.0));
+    pTrack->analysisFinished();
+    ProcessBuffer();
+
+    EXPECT_DOUBLE_EQ(200.0, m_pCuePoint->get());
+    EXPECT_DOUBLE_EQ(200.0, getCurrentSample());
+}
+
+TEST_F(CueControlTest, DontSeekOnLoadMainCue) {
     config()->set(ConfigKey("[Controls]", "CueRecall"),
             ConfigValue(static_cast<int>(SeekOnLoadMode::MainCue)));
     TrackPointer pTrack = createTestTrack();
     pTrack->setCuePoint(CuePosition(100.0));
 
+    // The Track should not follow if the cue point has been set manually in between.
     loadTrack(pTrack);
 
     EXPECT_DOUBLE_EQ(100.0, m_pCuePoint->get());
     EXPECT_DOUBLE_EQ(100.0, getCurrentSample());
 
     // Move cue like silence analysis does and check if track is following it
-    pTrack->setCuePoint(CuePosition(200.0));
+    pTrack->setCuePoint(CuePosition(400.0));
     pTrack->analysisFinished();
     ProcessBuffer();
 
-    EXPECT_DOUBLE_EQ(200.0, m_pCuePoint->get());
-    EXPECT_DOUBLE_EQ(200.0, getCurrentSample());
+    EXPECT_DOUBLE_EQ(400.0, m_pCuePoint->get());
+    EXPECT_DOUBLE_EQ(100.0, getCurrentSample());
 }
 
 TEST_F(CueControlTest, SeekOnLoadDefault_CueInPreroll) {
     config()->set(ConfigKey("[Controls]", "CueRecall"),
             ConfigValue(static_cast<int>(SeekOnLoadMode::MainCue)));
     TrackPointer pTrack = createTestTrack();
-    pTrack->setCuePoint(CuePosition(-100.0));
-
     loadTrack(pTrack);
 
-    EXPECT_DOUBLE_EQ(-100.0, m_pCuePoint->get());
-    EXPECT_DOUBLE_EQ(-100.0, getCurrentSample());
+    EXPECT_DOUBLE_EQ(0.0, m_pCuePoint->get());
+    EXPECT_DOUBLE_EQ(0.0, getCurrentSample());
```

Comment:
Why is this changed from `-100` to `0`? The purpose of this test was to verify that track is seeked to main cue when it is set in preroll (`cuePosition < 0`). This was a bug previously: https://bugs.launchpad.net/mixxx/+bug/1853919